pages/_app.js

```diff
@@ -1,10 +1,18 @@
+import { useState } from "react";
 import GlobalStyle from "../styles";
 
 export default function App({ Component, pageProps }) {
-	return (
-		<>
-			<GlobalStyle />
-			<Component {...pageProps} />
-		</>
-	);
+  //Hier States
+  const [productToBuy, setProduct] = useState();
```

Comment:
### Descriptive State Naming 
Rename `setProduct` to `setProductToBuy` for clarity

### State Initialization
Initialize `productToBuy` with `null` to explicitly indicate that it starts with no value.


`const [productToBuy, setProductToBuy] = useState(null);`